### ISSUE TITLE
WandB run names do not align with the local naming convention for model outputs, when using RayTune for hyperparameter tuning

### ISSUE THREAD

**Issue by rwilks-earthobs** [ultralytics/ultralytics#23000]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Hyperparameter Tuning

### Bug

**Context:**

I am running the hyperparameter tuning setup with Ray and WandB:

```
results_a = model_a.tune(data=train_val_yaml_a, 
                         project='my_project', name='my_run', 
                         use_ray=True, iterations=10, 
                         epochs=NUM_EPOCHS, imgsz=IMGSZ,  batch=BATCH_SIZE)
```

The **local output** from this tuning process is:

Current Directory
| - my_project
 __    | - my_run
 _____         | - _tune_2c2fc_00000
 _____         | - _tune_2c2fc_00001
 _____         | - _tune_2c2fc_00002 
 _____         | - ..... 
 _____         | - _tune_2c2fc_00009

Intuitively I would expect in wandb that these 10 runs would therefore run names which link to this:
my_run__0, my_run__1,..., my_run__9 (or some similar indexing). 

This would enable linking of files with local weights trained, which ideally I don't want to train again. 

**Issue:**
In wandb all runs are just given the name 'train'. I therefore can't link the best run which I find via wandb, to the local folder where the best weights will be stored. 

Thanks for your help

### Environment

Ultralytics 8.3.240 🚀 Python-3.12.11 torch-2.7.1+cu126 CUDA:0 (Tesla T4, 14914MiB)
Setup complete ✅ (128 CPUs, 503.5 GB RAM, 37.4/191.2 GB disk)

OS                     Linux-6.8.0-65-generic-x86_64-with-glibc2.35
Environment            Linux
Python                 3.12.11
Install                pip
Path                   /scratch/s2448783/miniforge3/envs/another_test/lib/python3.12/site-packages/ultralytics
RAM                    503.45 GB
Disk                   37.4/191.2 GB
CPU                    AMD EPYC 7542 32-Core Processor
CPU count              128
GPU                    Tesla T4, 14914MiB
GPU count              1
CUDA                   12.6

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.3>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 11.3.0>=7.1.2
pyyaml                 ✅ 6.0.2>=5.3.1
requests               ✅ 2.32.4>=2.23.0
scipy                  ✅ 1.16.0>=1.4.1
torch                  ✅ 2.7.1>=1.8.0
torch                  ✅ 2.7.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.22.1>=0.9.0
psutil                 ✅ 7.0.0>=5.8.0
polars                 ✅ 1.36.1>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18

### Minimal Reproducible Example

```
wandb.init()

results_a = model_a.tune(data=train_val_yaml_a, 
                         project='my_project', name='my_run', 
                         use_ray=True, iterations=10, 
                         epochs=NUM_EPOCHS, imgsz=IMGSZ,  batch=BATCH_SIZE)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rwilks-earthobs, thank you for your interest in Ultralytics 🚀! This is an automated response to help us route and diagnose your report faster—an Ultralytics engineer will also assist soon 🤖🔧. In the meantime, we recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. You’ve included a good start already—if possible, please expand it into a fully runnable snippet (including how `model_a` is created/loaded, any relevant Ray/W&B settings, and the exact expected vs actual W&B run name behavior). Also include any relevant console logs that show the run naming.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
With `use_ray=True` Ultralytics uses Ray Tune’s W&B callback, so calling `wandb.init()` yourself can collapse all trials into a single W&B run (often named `train`); please remove the manual init and let Ray/W&B create one run per trial as in the [Ray Tune + YOLO11 integration guide](https://docs.ultralytics.com/integrations/ray-tune/).  

```python
# no wandb.init() here
result_grid = model_a.tune(
    use_ray=True,
    data=train_val_yaml_a,
    project="my_project",
    name="my_run",
    iterations=10,
    epochs=NUM_EPOCHS,
    imgsz=IMGSZ,
    batch=BATCH_SIZE,
)

best = result_grid.get_best_result(metric="metrics/mAP50-95(B)", mode="max")
print(getattr(

**Comment by rwilks-earthobs** [comment]
Thanks for your reply. 

I also tried without wandb.init() and I was seeing the same behaviour, where each iteration in wandb was just given the name 'train' instead of run_name + iter (or something similar). I think it would be good for continuity across wandb and locally stored run artefacts that they link, in the same way as standard runs do. 

Thanks again

**Comment by glenn-jocher** [comment]
You’re right — with `use_ray=True` the `name=` you pass is used for the *Tune experiment folder*, but the per-trial YOLO training `name` isn’t propagated, so W&B ends up showing the default `train` display name; until that’s wired through (see the current `run_ray_tune()` behavior in the [tuner reference](https://docs.ultralytics.com/reference/utils/tuner/)), the reliable way to link W&B ↔ local artifacts is via the Ray trial directory (`Result.path`) / trial id (the `2c2fc` part in `_tune_2c2fc_00000`):

```python
from pathlib import Path

best = results_a.get_best_result(metric="metrics/mAP50-95(B)", mode="max")
trial_dir = Path(best.path)

print("Best trial dir:", trial_dir)
print("Best weights:", trial_dir / "weights" / "best.pt")
```
